### PR TITLE
[Fix Docs] Tutorial: Exploring Kubernetes CronJobs with Meshery Playground

### LINKED ISSUE #9833: [Docs] Tutorial: Exploring Kubernetes CronJobs with Meshery Playground
_This child issue is related to "[Playground] Learning Paths: Cloud Native tutorials with Meshery Playground #9832". See this issue for more context._

### Current State

Meshery Docs are in need of new tutorials with hands-on labs. In this tutorial, we will explore how to use Meshery Playground, an interactive live cluster environment, to perform hands-on labs for managing Kubernetes deployments. Meshery is a powerful tool that simplifies the adoption and operation of cloud and cloud native technologies. Meshery Playground provides a safe and isolated environment for learning and experimenting with these technologies.

### Desired State

1. Take the following sample and refactor in context of Meshery Playground. 
1. Instead of `kubectl` commands use Meshery UI and MeshMap as the primary mechanisms for users to perform tasks in the labs. Put CLI commands inside of `<details><summary></summary></details>` as an alternative reference.
1. Reference existing Meshery Designs from the Meshery Catalog both as starting point and/or as a finished example of the lab.
1. As needed, take screenshots or video recordings to enrich the content.

```yaml
---
layout: default
title: Exploring Kubernetes CronJobs
abstract: Learn Kubernetes CronJobs with Meshery Playground
permalink: guides/tutorials/
redirect_from: guides/tutorials/
type: guide
category: tutorial
language: en
abstract: "In this tutorial, we will explore how to use Meshery Playground, an interactive live cluster environment, to perform hands-on labs for managing Kubernetes CronJobs."
list: include
abstract: Meshery is the self-service engineering platform, enabling collaborative design and operation of cloud and cloud native infrastructure.
---

Introduction:
In this tutorial, we will explore Kubernetes CronJobs, a resource that allows you to run jobs periodically at specified intervals. We will use Meshery Playground, an interactive live cluster environment, to perform hands-on labs for working with CronJobs in Kubernetes.

Prerequisites:
- Basic understanding of Kubernetes concepts.
- Meshery Playground access. If you don't have an account, you can sign up at [Meshery Playground](https://meshery.layer5.io/play).

Lab Scenario: Scheduled Backups using CronJobs

Objective:
Learn how to use Kubernetes CronJobs to schedule and automate periodic backups of a sample application.

### Steps:

#### 1. **Accessing Meshery Playground:**
   - Log in to the [Meshery Playground](https://meshery.layer5.io/play) using your credentials.
   - Once logged in, navigate to the Meshery Playground dashboard.

#### 2. **Deploying a Sample Application:**
   - Deploy a simple application that needs periodic backups. You can use a sample web application or any other application of your choice.

\```bash
# Deploying a sample web application
mesheryctl pattern apply -f https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/application/deploying-wordpress/mysql-deployment.yaml
mesheryctl pattern apply -f https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/application/deploying-wordpress/wordpress-deployment.yaml
\```

<!-- 

Convey to user that MeshMap Designs are auto-saved. 

-->


#### 3. **Creating a CronJob for Backups:**

- Create a CronJob that runs a backup script at a specified interval to back up the application data.

<!--

Show user how to do this using MeshMap Designer to drag and drop components and configure them.

-->

   
<details>
<summary>
Alternate: Command line instructions
</summary>

\```yaml
# backup-cronjob.yaml
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: backup-cronjob
spec:
  schedule: "0 * * * *"  # Run every hour
  jobTemplate:
    spec:
      template:
        spec:
          containers:
          - name: backup-container
            image: backup-image:latest  # Replace with your backup script image
            # Add volume mounts and commands as needed for backup
          restartPolicy: OnFailure
\```

</details>

Apply the CronJob configuration:

\```bash
mesheryctl pattern apply -f backup-cronjob.yaml
\```

#### 4. **Verifying CronJob Execution:**
   - Monitor the execution of the CronJob and verify that backups are created at the specified intervals.


<!-- 

Show user how to use Views and filters in MeshMap Visualizer.

-->


#### 5. **Scaling and Updating CronJobs:**
   - Explore how to scale the CronJob or update its schedule by modifying the CronJob configuration.


<!-- 

Show user how to use Designs and components in MeshMap Designer.

-->


\```bash
kubectl edit cronjob backup-cronjob
\```

Use Meshery Playground to visualize the changes and observe the impact on the scheduled backups.

#### 6. **Clean-Up:**
   - Delete the CronJob and application resources after completing the lab.


<!-- 

Show user how to use Designs and components in MeshMap Designer.

-->



\```bash
kubectl delete cronjob backup-cronjob
kubectl delete -f https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/application/deploying-wordpress/mysql-deployment.yaml
kubectl delete -f https://raw.githubusercontent.com/kubernetes/website/master/content/en/examples/application/deploying-wordpress/wordpress-deployment.yaml
\```

#### 7. **Saving and Sharing**
   - Save your scenario in Meshery Playground for future reference.
   - Invite friends to collaborate in your design.

<!-- 

Show user how to make Design public and share with other users in MeshMap Designer.

-->

Share your scenario with other Meshery users or the community for collaborative learning.

### Conclusion
Congratulations! You've successfully completed the lab on exploring Kubernetes CronJobs using Meshery Playground. This hands-on experience has provided valuable insights into scheduling and automating tasks in Kubernetes using CronJobs. Explore more scenarios in the Meshery Playground to enhance your skills in cloud-native technologies.

{% include related-content.html type="tutorials" %}


```

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by sandramsc**
This PR fixes #9833

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 6e67bdb83eecf38ad1c87f56d9c8e4880a9d190b at: https://65b3ed90a4a5963d8bf862ba--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit c0aad55eafe72638d0bd0bb603a29bd5d28fd2c4 at: https://65b44771f4e8de2d868c3402--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a12d69fb4431091cb2ba4b7eaa9687891f014742 at: https://65b4f979b634208c0b35ab72--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3eaf42dddae1eab3002f1b1db7f8d08c0734a63e at: https://65c002d0d164b6a9f0beb12e--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8dd5b5226dea521ae4e05d6cfc46a4c59031ee4e at: https://65c6812a3c5c2f048c3df529--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 36f7b772da1ca51661466ce7dda572063d24a30f at: https://65c7eabf637dc6aae2a4fec3--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
Signal when ready for review.

**Comment by l5io** [comment]
🚀 Preview for commit cd6e25fa936732a985d3749f3b4161f9b95d2399 at: https://65d97aeac56064b943ba2c98--meshery-docs-preview.netlify.app

**Comment by sandramsc** [comment]
> Signal when ready for review.
@leecalcote 
Inquiring with @sangramrath  to see if public access is available for the tool used to create the annotations placed on these [screenshots](https://mesherydocs-git-issue9870-sangramrath.vercel.app/guides/tutorials/playground/kubernetes-pods.html) , so as to maintain a level of consistency (updating screenshots, then taking PR out of draft stage).

**Comment by sangramrath** [comment]
Hello @sandramsc I use a tool called Flameshot. https://flameshot.org/

File: docs/_includes/related-content.html
[the remark this answers is not in the document]
Comment (sandramsc) [reply docs/_includes/related-content.html:1]: Not sure what the ask is here, are you suggesting that they get added to the existing template while removing the old ones or?

**Comment by sangramrath** [comment]
I haven't ran the steps myself yet, but here are my initial thoughts:
- Step 4: we are saying URL import but the screenshot shows File Upload. Also if URL import, the URL link is missing.
- Probably we should use unique names when importing designs (add name or initials to the design). When searching, this makes for easy identification which is yours, else there would be many designs with same name. (there's a way to identify using info but that's longer process)
- Major heading 3: after step 8, it's step 1, should be 9 I guess?
- I am assuming the steps _bash mesheryctl pattern apply -f backup-cronjob.yaml_ and _bash kubectl edit cronjob backup-cronjob_ are alternate CLI instructions, if yes we should mention them. 
- A view from the visualize screen showing the cronjobs would be good at the end.
- May be we can crop the screenshots to only reflect the canvas area? where ever possible

Good job @sandramsc, sorry if it's a lot of comments :), do change only what's really necessary and of priority

**Comment by sandramsc** [comment]
> I haven't ran the steps myself yet, but here are my initial thoughts:
> 
>     * Step 4 (of Deploying a web application): we should provide a link to the URL or to download the design file (so users can upload)
> 
>     * Probably we should use unique names when importing designs (add name or initials to the design). When searching, this makes for easy identification which is yours, else there would be many designs with same name. (there's a way to identify using info but that's longer process)
> 
>     * Major heading 3: after step 8, it's step 1, should be 9 I guess?
> 
>     * I am assuming the steps _bash mesheryctl pattern apply -f backup-cronjob.yaml_ and _bash kubectl edit cronjob backup-cronjob_ are alternate CLI instructions, if yes we should mention them.
> 
>     * A view from the visualize screen showing the cronjobs would be good at the end.
> 
>     * May be we can crop the screenshots to only reflect the canvas area? where ever possible
> 
> 
> Good job @sandramsc, sorry if it's a lot of comments :), do change only what's really necessary and of priority

Hi @sangramrath thanks for the feedback. Are you able to make inline comments, as it might be better to associate each point of feedback with the section referenced, also perhaps we can have a chat about these on next weeks website call.

**Comment by sangramrath** [comment]
@sandramsc looks like I jumped the gun. Lee asked to check it out and I checked the last preview and left my comments. I will wait for the latest preview and leave inline comments. 
I assume the website call is Monday (looking at the history) around 6 PM my time, I am unable to promise but I will try.

**Comment by l5io** [comment]
🚀 Preview for commit 222074c1aa55ab18e605321a5aef4f6921963a6b at: https://65e5faf70fc94e06a5684e97--meshery-docs-preview.netlify.app

**Comment by sandramsc** [comment]
Closed unintentionally while trying to merge updates from upstream - reopening.

**Comment by iArchitSharma** [comment]
@sandramsc is this ready for review?

**Comment by l5io** [comment]
🚀 Preview for commit 150b8bd4ca6d81b5b7cad0eaedbc0154e96f7f9e at: https://6601750269f9eb1371ca795a--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@iArchitSharma, yes, I believe this is ready for review.

**Comment by l5io** [comment]
🚀 Preview for commit 58eab07d0bafe898a35db9b78c2490ba2c5f2f4d at: https://660d2671e94936be0be38b6f--meshery-docs-preview.netlify.app

**Review by iArchitSharma — APPROVED** [review]
LGTM Thanks @sandramsc

**Comment by l5io** [comment]
🚀 Preview for commit cd8bcca21cb21820b9ddd87041d3b08bcbf23fe9 at: https://66226a3c0787aa07ced77c18--meshery-docs-preview.netlify.app

**Pull request merged**